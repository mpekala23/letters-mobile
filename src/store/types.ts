import { UserState } from "./User/UserTypes";
import { NotifState } from "./Notif/NotifTypes";
import { ContactState } from "./Contact/ContactTypes";

export interface AppState {
  user: UserState;
  notif: NotifState;
  contact: ContactState;
}
