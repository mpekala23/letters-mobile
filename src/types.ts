// TODO: make this typing better
// probably want a specific type for state abbrevs
export interface Facility {
  name: string;
  type: PrisonTypes;
  address: string;
  city: string;
  state: string;
  postal: string;
  phone: string;
}

export enum ProfilePicTypes {
  Topbar = 'Topbar',
  Contact = 'Contact',
  SingleContact = 'SingleContact',
}

export enum Storage {
  RememberToken = 'Ameelio-Token',
}

export enum PrisonTypes {
  State = 'State',
  Federal = 'Federal',
  Jail = 'County',
}

export enum SupportFAQTypes {
  DeleteLetter = 'DeleteLetter',
  NotArrived = 'NotArrived',
  WrongReturnAddress = 'WrongReturnAddress',
  WrongMailingAddress = 'WrongMailingAddress',
  TrackingNumber = 'TrackingNumber',
  TrackingError = 'TrackingError',
  TalkToAmeelio = 'TalkToAmeelio',
}

export enum LetterStatus {
  Draft = 'Draft',
  Created = 'Created',
  Mailed = 'Mailed',
  InTransit = 'In Transit',
  InLocalArea = 'In Local Area',
  OutForDelivery = 'Out for Delivery',
  Delivered = 'Delivered',
}

export enum LetterTypes {
  Postcard = 'postcard',
  Letter = 'letter',
}

export enum DeliveryReportTypes {
  received = 'received',
  unsure = 'unsure',
  notYetReceived = 'notYetReceived',
  haveNotAsked = 'haveNotAsked',
  haveNotReceived = 'haveNotReceived',
}

export interface Letter {
  type: LetterTypes;
  status: LetterStatus;
  isDraft: boolean;
  recipientId: number;
  content: string;
  photo?: Photo;
  letterId?: number; // TODO: Once we have more info on this field and lob, use this more
  expectedDeliveryDate?: Date;
  dateCreated?: Date;
  trackingEvents?: LetterTrackingEvent[];
}

export interface LetterTrackingEvent {
  id: number;
  name: string;
  location: ZipcodeInfo;
  date: Date;
}

export type TopbarRouteAction = {
  enabled: boolean;
  text: string;
  action: () => void | Promise<void>;
  blocking?: boolean;
};

export interface ZipcodeInfo {
  zip: string;
  city: string;
  state: string;
  lat?: number;
  long?: number;
}

export interface Photo {
  type?: 'image' | string;
  width?: number;
  height?: number;
  uri: string;
}
