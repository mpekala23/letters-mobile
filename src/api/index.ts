import store from "@store";
import { Linking } from "react-native";
import { User, UserLoginInfo, UserRegisterInfo } from "@store/User/UserTypes";
import { dropdownError } from "@components/Dropdown/Dropdown.react";
import url from "url";
import { setItemAsync, getItemAsync, deleteItemAsync } from "expo-secure-store";
import { Storage } from "types";
import { loginUser, logoutUser } from "@store/User/UserActions";
import { setAdding, setExisting, clearContacts } from "store/Contact/ContactActions";
import "isomorphic-fetch";
import { Contact } from '@store/Contact/ContactTypes';

const MOCK_API_IP = process.env.MOCK_API_IP;
export const API_URL = "http://" + MOCK_API_IP + ":9000/api/";

export interface UserResponse {
  type: string;
  data: User;
}

export function fetchTimeout(
  url: string,
  options: object,
  timeout = 3000
): Promise<Response> {
  return Promise.race([
    fetch(url, options),
    new Promise<Response>((_, reject) =>
      setTimeout(() => reject(new Error("timeout")), timeout)
    ),
  ]);
}

export async function saveToken(token: string) {
  return await setItemAsync(Storage.RememberToken, token);
}

export async function deleteToken() {
  return await deleteItemAsync(Storage.RememberToken);
}

export async function loginWithToken() {
  try {
    const rememberToken = await getItemAsync(Storage.RememberToken);
    if (!rememberToken) {
      throw Error("Cannot load token");
    }
    const response = await fetchTimeout(url.resolve(API_URL, "login/token"), {
      method: "POST",
      headers: {
        Accept: "application/json",
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        token: rememberToken,
      }),
    });
    const body = await response.json();
    if (body.status === "ERROR") throw Error("Invalid token");
    const userData: User = {
      id: body.data.id,
      firstName: body.data.first_name,
      lastName: body.data.last_name,
      email: body.data.email,
      phone: body.data.phone,
      address1: body.data.addr_line_1,
      address2: body.data.addr_line_2 || "",
      country: body.data.country,
      postal: body.data.postal,
      city: body.data.city,
      state: body.data.state,
    };
    store.dispatch(loginUser(userData));
    return userData;
  } catch (err) {
    store.dispatch(logoutUser());
    throw Error(err);
  }
}

export async function login(cred: UserLoginInfo) {
  const response = await fetchTimeout(url.resolve(API_URL, "login"), {
    method: "POST",
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      email: cred.email,
      password: cred.password,
    }),
  });
  const body = await response.json();
  if (body.status == "ERROR") {
    throw Error(body.message);
  }
  if (cred.remember) {
    try {
      // TODO: Once documentation is complete, ensure that this is wherere the info will be stored
      await saveToken(body.data.token);
    } catch (err) {
      dropdownError(
        "Storage",
        "Unable to save login credentials for next time"
      );
    }
  }
  const userData: User = {
    id: body.data.id,
    firstName: body.data.first_name,
    lastName: body.data.last_name,
    email: body.data.email,
    phone: body.data.phone,
    address1: body.data.addr_line_1,
    address2: body.data.addr_line_2 || "",
    country: body.data.country,
    postal: body.data.postal,
    city: body.data.city,
    state: body.data.state,
  };
  store.dispatch(loginUser(userData));
  return userData;
}

export async function logout() {
  store.dispatch(logoutUser());
  store.dispatch(clearContacts());
  return await deleteToken();
}

export async function register(data: UserRegisterInfo) {
  const response = await fetchTimeout(url.resolve(API_URL, "register"), {
    method: "POST",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      email: data.email,
      first_name: data.firstName,
      last_name: data.lastName,
      phone: data.phone,
      addr_line_1: data.address1,
      addr_line_2: data.address2,
      country: data.country,
      postal: data.postal,
      city: data.city,
      state: data.state,
    }),
  });
  const body = await response.json();
  if (body.status == "ERROR") {
    throw Error(body.message);
  }
  if (data.remember) {
    try {
      // TODO: Once documentation is complete, ensure that this is wherere the info will be stored
      await saveToken(body.data.token);
    } catch (err) {
      dropdownError(
        "Storage",
        "Unable to save login credentials for next time"
      );
    }
  }
  const userData: User = {
    id: body.data.id,
    firstName: body.data.first_name,
    lastName: body.data.last_name,
    email: body.data.email,
    phone: body.data.phone,
    address1: body.data.addr_line_1,
    address2: body.data.addr_line_2 || null,
    country: body.data.country,
    postal: body.data.postal,
    city: body.data.city,
    state: body.data.state,
  };
  store.dispatch(loginUser(userData));
  return userData;
}

export async function getFacilities(text: string) {}

export async function addContact(data: {}) {
  const response = await fetchTimeout(url.resolve(API_URL, 'contacts'), {
    method: 'POST',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(data),
  });
  const body = await response.json();
  if (body.type == 'ERROR') {
    throw Error(body.data);
  }
  const contactData: Contact = {
    firstName: body.data.first_name,
    lastName: body.data.last_name,
    inmateNumber: body.data.inmate_number,
    state: body.data.state,
    relationship: body.data.relationship,
    facility: body.data.facility,
  };
  const { existing } = store.getState().contact;
  // Add contact to existing list of contacts
  existing.push(contactData);
  store.dispatch(setExisting(existing));
  store.dispatch(
    setAdding({
      state: '',
      firstName: '',
      lastName: '',
      inmateNumber: '',
      relationship: '',
      facility: null,
    })
  );
  return existing;
}

export async function facebookShare(shareUrl: string) {
  const supportedUrl = await Linking.canOpenURL(shareUrl);
  if (supportedUrl) {
    await Linking.openURL(shareUrl);
  } else {
    throw Error('Share Url not supported');
  }
}
