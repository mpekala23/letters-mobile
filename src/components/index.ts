import AdjustableText from './Text/AdjustableText.react';
import Alert from './Alert/Alert.react';
import AsyncImage from './AsyncImage/AsyncImage.react';
import Button from './Button/Button.react';
import CategoryCard from './Card/CategoryCard.react';
import ContactSelectorCard from './Card/ContactSelectorCard.react';
import CreditsCard from './Card/CreditsCard.react';
import ComposeHeader from './ComposeHeader/ComposeHeader.react';
import ComposeTools from './ComposeTools/ComposeTools.react';
import DeliveryStatusCard from './Card/DeliveryStatusCard.react';
import DisplayImage from './DisplayImage/DisplayImage.react';
import Dropdown from './Dropdown/Dropdown.react';
import DynamicPostcard from './DynamicPostcard/DynamicPostcard.react';
import EditablePostcard from './EditablePostcard/EditablePostcard.react';
import GenericCard from './Card/GenericCard.react';
import GrayBar from './GrayBar/GrayBar.react';
import HeaderLeft from './Topbar/HeaderLeft.react';
import HeaderTitle from './Topbar/HeaderTitle.react';
import HeaderRight from './Topbar/HeaderRight.react';
import Icon from './Icon/Icon.react';
import Input from './Input/Input.react';
import KeyboardAvoider from './KeyboardAvoider/KeyboardAvoider.react';
import LetterOptionCard from './Card/LetterOptionCard.react';
import LetterStatusCard from './Card/LetterStatusCard.react';
import LetterTracker from './LetterTracker/LetterTracker.react';
import ListItem from './ListItem/ListItem.react';
import MailingAddressPreview from './MailingAddressPreview/MailingAddressPreview.react';
import MemoryLaneCard from './Card/MemoryLaneCard.react';
import Picker, { PickerRef } from './Picker/Picker.react';
import PicUpload from './PicUpload/PicUpload.react';
import PostcardTools from './PostcardTools/PostcardTools.react';
import PrisonCard from './Card/PrisonCard.react';
import ProfilePic from './ProfilePic/ProfilePic.react';
import StaticPostcard from './StaticPostcard/StaticPostcard.react';
import Statusbar from './Statusbar/Statusbar.react';
import StickerManager from './StickerManager/StickerManager.react';

export {
  AdjustableText,
  Alert,
  AsyncImage,
  Button,
  CategoryCard,
  ContactSelectorCard,
  CreditsCard,
  ComposeHeader,
  ComposeTools,
  DeliveryStatusCard,
  DisplayImage,
  Dropdown,
  DynamicPostcard,
  EditablePostcard,
  GenericCard,
  GrayBar,
  HeaderLeft,
  HeaderTitle,
  HeaderRight,
  Icon,
  Input,
  KeyboardAvoider,
  LetterOptionCard,
  LetterStatusCard,
  LetterTracker,
  ListItem,
  MailingAddressPreview,
  MemoryLaneCard,
  Picker,
  PicUpload,
  PostcardTools,
  PrisonCard,
  ProfilePic,
  StaticPostcard,
  Statusbar,
  StickerManager,
};

export type { PickerRef };
