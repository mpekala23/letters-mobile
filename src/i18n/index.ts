import i18n from 'i18n-js';
import * as Localization from 'expo-localization';

import en from './locales/en.json';
import es from './locales/es.json';

i18n.defaultLocale = 'en';
i18n.fallbacks = true;
i18n.translations = { en, es };
i18n.locale = Localization.locale;
export default i18n;
