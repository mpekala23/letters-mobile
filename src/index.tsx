/* eslint-disable global-require */
import React from 'react';
import { Provider } from 'react-redux';
import store, { persistor } from '@store';
import Navigator from '@navigations';
import { Alert, Dropdown, Statusbar } from '@components';
import { loginWithToken } from '@api';
import { PersistGate } from 'redux-persist/integration/react';
import * as Font from 'expo-font';
import { AppLoading } from 'expo';
import { setCustomText } from 'react-native-global-props';
import * as Segment from 'expo-analytics-segment';
import * as Sentry from 'sentry-expo';
import Constants from 'expo-constants';
import { isProduction } from '@utils';

const customFonts = {
  'Poppins-Light': require('./assets/fonts/Poppins-Light.ttf'),
  'Poppins-Light-Italic': require('./assets/fonts/Poppins-LightItalic.ttf'),
  'Poppins-Regular': require('./assets/fonts/Poppins-Regular.ttf'),
  'Poppins-Regular-Italic': require('./assets/fonts/Poppins-Italic.ttf'),
  'Poppins-Medium': require('./assets/fonts/Poppins-Medium.ttf'),
  'Poppins-Medium-Italic': require('./assets/fonts/Poppins-MediumItalic.ttf'),
  'Poppins-SemiBold': require('./assets/fonts/Poppins-SemiBold.ttf'),
  'Poppins-SemiBold-Italic': require('./assets/fonts/Poppins-SemiBoldItalic.ttf'),
  'Poppins-Bold': require('./assets/fonts/Poppins-Bold.ttf'),
  'Poppins-Bold-Italic': require('./assets/fonts/Poppins-BoldItalic.ttf'),
};

const customTextProps = {
  style: {
    fontFamily: 'Poppins-Regular',
  },
};

export interface State {
  fontsLoaded: boolean;
}

export default class App extends React.Component<null, State> {
  constructor(props: null) {
    super(props);
    this.state = { fontsLoaded: false };
  }

  async componentDidMount(): Promise<void> {
    this.loadFontsAsync();
    if (isProduction()) {
      Sentry.init({
        dsn:
          'https://15e7f78b44064e1eb57afb5c8a239122@o434922.ingest.sentry.io/5392541',
        release: process.env.APP_RELEASE || '1.0.11',
      });
    }

    const { androidWriteKey, iosWriteKey } = App.getSegmentWriteKeys();
    Segment.initialize({ androidWriteKey, iosWriteKey });
    Segment.trackWithProperties('App Open', {
      'App Version': process.env.APP_VERSION,
      'Native Build Version': Constants.nativeBuildVersion,
    });
    try {
      await loginWithToken();
    } catch (err) {
      /* Unable to load token */
    }
  }

  componentWillUnmount(): void {
    Segment.track('App Closed');
  }

  static getSegmentWriteKeys(): Record<string, string> {
    if (isProduction())
      return {
        androidWriteKey: 'cveBC1HNczxB1HgrrquX8zjjfRAapEmx',
        iosWriteKey: 'EryEQcrwG2YGcPKFsPz8AGUTZ9Rdcqvi',
      };
    // On development or staging channels
    return {
      androidWriteKey: 'skQ1SzNOGHiOF2o5vkOCZzhl4QXykseD',
      iosWriteKey: 'emYpyC3ipSbi6XDHqaFn6mGuad2vn6Xy',
    };
  }

  async loadFontsAsync(): Promise<void> {
    await Font.loadAsync(customFonts);
    setCustomText(customTextProps);
    this.setState({ fontsLoaded: true });
  }

  render(): JSX.Element {
    if (this.state.fontsLoaded) {
      return (
        <Provider store={store}>
          <PersistGate persistor={persistor}>
            <Statusbar />
            <Navigator />
            <Alert />
            <Dropdown />
          </PersistGate>
        </Provider>
      );
    }
    return <AppLoading />;
  }
}
