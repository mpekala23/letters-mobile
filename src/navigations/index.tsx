import React, { useState } from 'react';
import { connect } from 'react-redux';
import { AppState } from '@store/types';
import { AuthInfo, UserState } from '@store/User/UserTypes';
import { navigationRef, navigate } from '@utils';
import { setProfile } from '@components/Topbar';
import { NavigationContainer } from '@react-navigation/native';
import HomeIcon from '@assets/navigation/Home';
import StoreIcon from '@assets/navigation/Store';

import i18n from '@i18n';
import {
  Screens,
  AuthStackParamList,
  AppStackParamList,
  getDetailsFromRouteName,
  Tabs,
} from '@utils/Screens';
import { SplashScreen } from '@views';
import { GestureResponderEvent } from 'react-native';
import { RootTab } from './Navigators';

import Auth from './Auth';
import Home from './Home';
import Store from './Store';
import TabIcon from './TabIcon.react';

export { navigationRef, navigate };

export type RootStackParamList = AuthStackParamList & AppStackParamList;

export interface Props {
  authInfo: AuthInfo;
  userState: UserState;
}

const NavigatorBase: React.FC<Props> = (props: Props) => {
  const [tabsVisible, setTabsVisible] = useState(true);

  // Determine which views should be accessible
  let screens;
  if (
    props.authInfo.isLoadingToken ||
    (props.authInfo.isLoggedIn && !props.authInfo.isLoaded)
  ) {
    screens = <RootTab.Screen name={Tabs.Splash} component={SplashScreen} />;
  } else if (!props.authInfo.isLoggedIn) {
    screens = <RootTab.Screen name={Tabs.Auth} component={Auth} />;
  } else {
    screens = (
      <>
        <RootTab.Screen
          name={Tabs.Home}
          component={Home}
          options={{
            tabBarButton: (tabProps) => {
              return (
                <TabIcon
                  name={i18n.t('Navigation.home')}
                  svg={HomeIcon}
                  onPress={(e: GestureResponderEvent) => {
                    if (tabProps.onPress) {
                      tabProps.onPress(e);
                    }
                  }}
                />
              );
            },
          }}
        />
        <RootTab.Screen
          name={Tabs.Store}
          component={Store}
          options={{
            tabBarButton: (tabProps) => {
              return (
                <TabIcon
                  name={i18n.t('Navigation.store')}
                  svg={StoreIcon}
                  onPress={(e: GestureResponderEvent) => {
                    if (tabProps.onPress) {
                      tabProps.onPress(e);
                    }
                  }}
                />
              );
            },
          }}
        />
      </>
    );
  }

  return (
    <NavigationContainer
      ref={navigationRef}
      onStateChange={() => {
        const name = navigationRef.current?.getCurrentRoute()?.name as Screens;
        if (name) {
          setProfile(!!getDetailsFromRouteName(name).profile);
          if (
            getDetailsFromRouteName(name).tabsVisible === undefined ||
            getDetailsFromRouteName(name).tabsVisible === true
          )
            setTabsVisible(true);
          else setTabsVisible(false);
        } else {
          setProfile(true);
          setTabsVisible(true);
        }
      }}
    >
      <RootTab.Navigator
        screenOptions={({ route }) => ({
          tabBarVisible:
            route.name !== Tabs.Auth &&
            route.name !== Tabs.Splash &&
            tabsVisible,
          tabBarVisibilityAnimationConfig: {},
        })}
        tabBarOptions={{
          style: { height: 64 },
          keyboardHidesTabBar: true,
        }}
      >
        {screens}
      </RootTab.Navigator>
    </NavigationContainer>
  );
};

const mapStateToProps = (state: AppState) => ({
  authInfo: state.user.authInfo,
  userState: state.user,
});
const Navigator = connect(mapStateToProps)(NavigatorBase);

export default Navigator;
