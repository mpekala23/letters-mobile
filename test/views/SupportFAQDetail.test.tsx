import React from 'react';
import { SupportFAQDetailScreen } from '@views';
import { render, toJSON, fireEvent } from '@testing-library/react-native';
import { SupportFAQTypes } from '../../src/types';

const setup = (issueOverrides: Record<string, unknown>) => {
  const navigation = { navigate: jest.fn() };
  const route = {
    params: { issue: '', ...issueOverrides },
  };
  return {
    navigation,
    route,
    ...render(<SupportFAQDetailScreen navigation={navigation} route={route} />),
  };
};

describe('Support FAQ Detail screen', () => {
  it('should match snapshot', () => {
    const { container } = setup(<SupportFAQDetailScreen />);
    const tree = toJSON(container);
    expect(tree).toMatchSnapshot();
  });

  it('no call-to-action for DeleteLetter', () => {
    const { getByTestId } = setup({
      issue: SupportFAQTypes.DeleteLetter,
    });
    expect(getByTestId('callToActionButton').props.children).toBe(null);
  });

  it('no call-to-action for NotArrived', () => {
    const { getByTestId } = setup({
      issue: SupportFAQTypes.NotArrived,
    });
    expect(getByTestId('callToActionButton').props.children).toBe(null);
  });

  it('no call-to-action for TrackingNumber', () => {
    const { getByTestId } = setup({
      issue: SupportFAQTypes.TrackingNumber,
    });
    expect(getByTestId('callToActionButton').props.children).toBe(null);
  });

  it('should navigate to update contact screen when WrongMailingAddress button is pressed', () => {
    const { navigation, getByText } = setup({
      issue: SupportFAQTypes.WrongMailingAddress,
    });
    fireEvent.press(getByText('Update address'));
    expect(navigation.navigate).toHaveBeenCalledWith('UpdateContact');
  });

  // TO-DO: Should navigate to update profile screen when WrongReturnAddress button is pressed
});
