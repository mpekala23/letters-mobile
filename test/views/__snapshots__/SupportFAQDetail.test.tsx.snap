// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Support FAQ Detail screen should match snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flex": 1,
        "flexDirection": "column",
        "padding": 16,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Poppins-Bold",
          },
          Object {
            "color": "#282828",
            "fontSize": 20,
            "paddingBottom": 20,
            "paddingTop": 20,
          },
        ]
      }
    >
      Talk to someone at Ameelio.
    </Text>
    <Text
      style={
        Object {
          "color": "#282828",
          "fontSize": 16,
          "paddingBottom": 30,
          "paddingTop": 20,
          "textAlign": "center",
          "width": "90%",
        }
      }
    >
      We’re happy to talk and we’d be more than glad to help however we can!
    </Text>
    <View
      style={
        Object {
          "width": "100%",
        }
      }
      testID="callToActionButton"
    >
      <TouchableOpacity
        activeOpacity={0.7}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#FF7171",
              "borderRadius": 16,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginVertical": 5,
              "paddingHorizontal": 14,
            },
            Object {},
            Object {
              "backgroundColor": "#FC7272",
              "borderRadius": 15,
              "height": 50,
              "width": "100%",
            },
            Object {},
            Object {
              "elevation": 4,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 5,
                "width": 0,
              },
              "shadowOpacity": 0.22,
            },
          ]
        }
        testID="clickable"
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins-Medium",
              },
              Object {
                "color": "white",
                "fontSize": 20,
                "fontWeight": "700",
                "padding": 6,
              },
              Object {},
              Object {
                "fontSize": 18,
              },
              Object {},
              Object {
                "opacity": 1,
              },
            ]
          }
        >
          Reach out to support
        </Text>
      </TouchableOpacity>
    </View>
  </View>
</View>
`;
